### PR TITLE
tutorial section refactor

### PR DESCRIPTION
**Pull request by AroosaIjaz**
As the doc-refactor PR has been merged, the tutorial section is presented as a gallery on a single page with clean and concise TOC 

**Description of the Change:**

As discussed in the doc refactor PR, all tutorials have been put neatly on a single page:
  
1. all tutorials are python files. 
2. tutorials have been divided into sections for both smooth transitions and easy access from the TOC. the TOC is not cluttered and any particular topic can be easily chosen. 
3. most of the tutorials are compiled on the run when docs are compiled. 

**Benefits:**

Has examples and tutorials for everyone from young beginners to advanced users. 
All previous PRs on this topic can be removed if everyone likes this format.

**issues to be fixed**
1. The strawberry field plugin is currently not supporting `NumberOperator` observable so I skipped running the files where it is being used for now. 
2. some figures in the QML section (vqe) are not being plotted properly.

### CODE REVIEW

File: doc/tutorials.rst [hunk 1]
```diff
@@ -6,12 +6,52 @@
 Tutorials
 =========
 
-:html:`<h3>Learn PennyLane</h3>`
+PennyLane basics
+~~~~~~~~~~~~~~~~~~~
 
+These step-by-step tutorials provide a gentle introduction to the basics of PennyLane. In the last two examples, we put together everything we learn from the basic tutorials and consider PennyLane's version of the "Hello world!" example - qubit rotation and qumode displacement.
```
Comment (mariaschuld) [inline doc/tutorials.rst]:
```suggestion
These step-by-step tutorials provide a gentle introduction to the basics of PennyLane. In the last two examples, we put everything together and consider PennyLane's version of the "Hello world!" example - qubit rotation and the Gaussian transformation of qumode displacement.
```

Just to shorten and to link the text to the title of the tutorial.

File: doc/tutorials.rst [hunk 2]
```diff
@@ -20,37 +60,119 @@ plugins, and devices, via simple and easy-to-follow examples.
...
-    :figure: ../examples/figures/gaussian_transformation.png
+    :figure: ../examples/figures/gaussiancircuit.png
     :description: :ref:`gaussian_transformation`
 
+:html:`<div style='clear:both'></div>`
+
+Classical interfaces
+~~~~~~~~~~~~~~~~~~~~~
+
+The following tutorials take an in-depth look at one and the same Ising model tutorial, implemented across different interfaces offered by PennyLane.
```
Comment (mariaschuld) [inline doc/tutorials.rst]:
```suggestion
The following tutorials demonstrate how to optimise the energy of a simple Ising model using different interfaces offered by PennyLane. 
```
I was very confused as to where the Ising model tutorial was mentioned before. Ok to change?

File: doc/tutorials.rst [hunk 3]
```diff
@@ -20,37 +60,119 @@ plugins, and devices, via simple and easy-to-follow examples.
...
+The most powerful design feature of PennyLane is that it allows the user to run hybrid computations on a variety of different simulators and real quantum hardware. The following tutorials introduce the notion of hybrid computation by combining several PennyLane plugins.
+
+
+.. toctree::
+    :hidden:
+    :maxdepth: 1
+
+    tutorials/pennylane_plugins_hybrid
+    tutorials/pennylane_pytorch_noise
+    tutorials/pennylane_run_state_preparation
```
Comment (mariaschuld) [inline doc/tutorials.rst]: I'm getting a 

TypeError: Can't instantiate abstract class QVMDevice with abstract methods observables

when building the docs. The error shows in the tutorial after the line "dev = qml.device("forest.qvm", device="3q-pyqvm", shots=1000)"

File: doc/tutorials.rst [hunk 4]
```diff
@@ -20,37 +60,119 @@ plugins, and devices, via simple and easy-to-follow examples.
...
+    :figure: ../examples/figures/pytorch.png
+    :description: :ref:`isingmodel_PyTorch`
+
+:html:`<div style='clear:both'></div>`
+
+
+Plugins and hybrid computation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The most powerful design feature of PennyLane is that it allows the user to run hybrid computations on a variety of different simulators and real quantum hardware. The following tutorials introduce the notion of hybrid computation by combining several PennyLane plugins.
```
Comment (mariaschuld) [inline doc/tutorials.rst]: Are they really? The "pytorch_noise" tutorial integrates the forest plugin with the pytorch interface, and the "state_prep" one also uses forest only. 

I propose to change the summary a bit, so people don't get confused...

File: doc/tutorials.rst [hunk 5]
```diff
@@ -20,37 +60,119 @@ plugins, and devices, via simple and easy-to-follow examples.
...
+    :tooltip: Make a Quantum Neural Network in a single line of code.
+    :figure: ../examples/figures/cvqnn_example.png
+    :description: :ref:`templates_CV`
+
+:html:`<div style='clear:both'></div>`
+
+Quantum machine learning algorithms
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+PennyLane is desgined to do machine learning with quantum computers. These examples show PennyLane's implementation of some of the well-known algorithms from the recent research in quantum machine learning. This is, however, just to showcase how quantum machine learning with PennyLane *could* look like. In principle *any* technique where a quantum circuit is optimized and potentially run on a real device can be implemented in PennyLane.
```
Comment (mariaschuld) [inline doc/tutorials.rst]: Just so we don't limit ourselves. Maybe in future optimisation without ML will become more important.

```suggestion
One goal of PennyLane is to enable machine learning with quantum computers. These examples show PennyLane's implementation of some of the well-known algorithms from the recent research in quantum machine learning. This is, however, just to showcase how quantum machine learning with PennyLane *could* look like. In principle *any* technique where a quantum circuit is optimized and potentially run on a real device can be implemented in PennyLane.  
```

**Comment by AroosaIjaz** [comment]
> One overall comment: When I try to download the notebooks, it opens a new tab in my browser with lots of funny code instead, like when opening a notebook as text file. Is that issue known to you?

yes, it works fine when you save them (right-click--> save as) and use them. I agree it will be better if it is directly downloaded though, @josh146.

**Comment by josh146** [comment]
> yes, it works fine when you save them (right-click--> save as) and use them. I agree it will be better if it is directly downloaded though

Oh that's weird - it works fine on the recently built read the docs version (see https://pennylane.readthedocs.io/en/latest/tutorials/pennylane_run_qubit_rotation.html#qubit-rotation).

Does it fail on all tutorial pages in this PR, or just some?

**Comment by AroosaIjaz** [comment]
> > yes, it works fine when you save them (right-click--> save as) and use them. I agree it will be better if it is directly downloaded though
> 
> Oh that's weird - it works fine on the recently built read the docs version (see https://pennylane.readthedocs.io/en/latest/tutorials/pennylane_run_qubit_rotation.html#qubit-rotation).
> 
> Does it fail on all tutorial pages in this PR, or just some?

It fails for all tutorial pages.

File: doc/tutorials.rst (hunk 3 above)
Comment (AroosaIjaz) [reply doc/tutorials.rst]: @mariaschuld did this issue workout?

File: doc/tutorials.rst [hunk 6]
```diff
@@ -40,17 +123,56 @@ plugins, and devices, via simple and easy-to-follow examples.
...
+    :tooltip: Make a strongly entangling circuit in a single line of code.
+    :figure: ../examples/figures/slsec_example.png
+    :description: :ref:`templates_DV`
+
+.. customgalleryitem::
+    :tooltip: Make a Quantum Neural Network in a single line of code.
+    :figure: ../examples/figures/cvqnn_example.png
+    :description: :ref:`templates_CV`
+
+:html:`<div style='clear:both'></div>`
```
[the remark this answers is not in the document]
Comment (AroosaIjaz) [reply doc/tutorials.rst]: hm this does not sound bad. I will go for this approach in the new PRs :)

**Comment by AroosaIjaz** [comment]
As @josh146  suggested, I will be closing this PR and submitting many small PRs instead as they tend to move along faster. 

Thank you for your review :)

**Pull request closed without merging**